src/gui/qgsdatasourcemanagerdialog.cpp

```diff
@@ -90,7 +110,8 @@ QgsDataSourceManagerDialog::~QgsDataSourceManagerDialog()
 
 void QgsDataSourceManagerDialog::openPage( const QString &pageName )
 {
-  const int pageIdx = mPageNames.indexOf( pageName );
+  // TODO -- this is actually using provider keys, not provider names!
```

Comment:
Still TODO?

Reply:
yep -- it's the stable api which is wrong, and I'm not confident to change it.

Reply:
by that I mean I'm not confidently that if we switch to using keys here internally now we won't break a plugin

Reply:
@nyalldawson , fair enough, just wanted to make sure it was a TODO for posterity and not for this PR :)